src/app-layout/index.tsx

```diff
@@ -51,9 +62,36 @@ const AppLayout = React.forwardRef(
       drawersOverflow: i18n('ariaLabels.drawersOverflow', rest.ariaLabels?.drawersOverflow),
       drawersOverflowWithBadge: i18n('ariaLabels.drawersOverflowWithBadge', rest.ariaLabels?.drawersOverflowWithBadge),
     };
+    const { navigationOpen: defaultNavigationOpen, ...restDefaults } = applyDefaults(
+      contentType,
+      { maxContentWidth, minContentWidth },
+      isRefresh
+    );
+
+    const [navigationOpen = false, setNavigationOpen] = useControllable(
+      controlledNavigationOpen,
+      controlledOnNavigationChange,
+      isMobile ? false : defaultNavigationOpen,
+      { componentName: 'AppLayout', controlledProp: 'navigationOpen', changeHandler: 'onNavigationChange' }
+    );
+    const onNavigationChange = (event: NonCancelableCustomEvent<AppLayoutProps.ChangeDetail>) => {
+      setNavigationOpen(event.detail.open);
+      controlledOnNavigationChange?.(event);
+    };
```

Comment:
Apply controllability behavior here once and all code down in this component assumes `navigationOpen` and `onNavigationChange` to be always controlled